starter-code/routes/auth.js

```diff
@@ -0,0 +1,116 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const User = require("../models/auth");
+
+/* GET home page */
+router.get("/signup", (req, res, next) => {
+  const { error } = req.query;
+  switch (error) {
+    case "Empty":
+      res.render("auth/signup", { error: "Please fill in all the fields" });
+      break;
+    case "Exists":
+      res.render("auth/signup", { error: "Username already exists" });
+      break;
+    default:
+      res.render("auth/signup");
+  }
+});
+
+router.post("/signup", (req, res, next) => {
```

Comment:
I like the way how you handled all scenarios and threw the errors to your login page/signup page. ✅